boot/freeldr/freeldr/include/fs/ext.h

```diff
@@ -0,0 +1,239 @@
+ */
+
+#pragma once
+#include <pshpack1.h>
+
+#define EXT_SUPERBLOCK_MAGIC 0xEF53
+#define EXT_DYNAMIC_REVISION 1
+#define EXT_DEFAULT_INODE_SIZE 128
+#define EXT_DEFAULT_GROUP_DESC_SIZE 32
+
+#define EXT_DIR_ENTRY_MAX_NAME_LENGTH 255
+
+struct _ExtSuperBlock
+{
+    /* SuperBlock Information Ext2 */
+    ULONG InodesCount;
+    ULONG BlocksCountLo;
+    ULONG RBlocksCountLo;
+    ULONG FreeBlocksCountLo;
+    ULONG FreeInodesCount;
+    ULONG FirstDataBlock;
+    ULONG LogBlockSize;
+    LONG LogFragSize;
+    ULONG BlocksPerGroup;
+    ULONG FragsPerGroup;
+    ULONG InodesPerGroup;
+    ULONG MTime;
+    ULONG WTime;
+    USHORT MntCount;
+    USHORT MaxMntCount;
+    USHORT Magic;
+    USHORT State;
+    USHORT Errors;
+    USHORT MinorRevisionLevel;
+    ULONG LastCheck;
+    ULONG CheckInterval;
+    ULONG CreatorOS;
+    ULONG RevisionLevel;
+    USHORT DefResUID;
+    USHORT DefResGID;
+
+    /* SuperBlock Information Ext3 */
+    ULONG FirstInode;
+    USHORT InodeSize;
+    USHORT BlockGroupNr;
+    ULONG FeatureCompat;
+    ULONG FeatureIncompat;
+    ULONG FeatureROCompat;
+    UCHAR UUID[16];
+    char VolumeName[16];
+    char LastMounted[64];
+    ULONG AlgorithmUsageBitmap;
+    UCHAR PreallocBlocks;
+    UCHAR PreallocDirBlocks;
+    USHORT ReservedGdtBlocks;
+    char JournalUUID[16];
+    ULONG JournalInum;
+    ULONG JournalDev;
+    ULONG LastOrphan;
+    ULONG HashSeed[4];
+    UCHAR DefHashVersion;
+    UCHAR JournalBackupType;
+    USHORT GroupDescSize;
+    UCHAR Reserved[768];
+};
+
+struct _ExtGroupDescriptor
+{
+    ULONG BlockBitmap;
+    ULONG InodeBitmap;
+    ULONG InodeTable;
+    USHORT FreeBlocksCount;
+    USHORT FreeInodesCount;
+    USHORT UsedDirsCount;
+};
+
+struct _Ext4ExtentHeader
+{
+    USHORT Magic;
+    USHORT Entries;
+    USHORT Max;
+    USHORT Depth;
+    ULONG Generation;
+};
+
+struct _Ext4ExtentIdx
+{
+    ULONG Block;
+    ULONG Leaf;
+    USHORT LeafHigh;
+    USHORT Unused;
+};
+
+struct _Ext4Extent
+{
+    ULONG Block;
+    USHORT Length;
+    USHORT StartHigh;
+    ULONG Start;
+};
+
+struct _ExtInode
+{
+    USHORT Mode;
+    USHORT UID;
+    ULONG Size;
+    ULONG Atime;
+    ULONG Ctime;
+    ULONG Mtime;
+    ULONG Dtime;
+    USHORT GID;
+    USHORT LinksCount;
+    ULONG BlocksCount;
+    ULONG Flags;
+    ULONG OSD1;
+    union
+    {
+        char SymLink[60];
+        struct
+        {
+            ULONG DirectBlocks[12];
+            ULONG IndirectBlock;
+            ULONG DoubleIndirectBlock;
+            ULONG TripleIndirectBlock;
+        } Blocks;
+        struct _Ext4ExtentHeader ExtentHeader;
+    };
+    ULONG Generation;
+    ULONG FileACL;
+    ULONG DirACL;
+    ULONG FragAddress;
+    ULONG OSD2[3];
+};
+
+struct _ExtDirEntry
+{
+    ULONG Inode;
+    USHORT EntryLen;
+    UCHAR NameLen;
+    UCHAR FileType;
+    char Name[EXT_DIR_ENTRY_MAX_NAME_LENGTH];
+};
+
+#include <poppack.h>
+
+typedef struct _ExtSuperBlock EXT_SUPER_BLOCK, *PEXT_SUPER_BLOCK;
```

Comment:
Since you've refactored the structures above (with respect to the original grub ones), I would suggest you directly define them as: `typedef struct _ExtBlaBla { ... } EXT_BLABLA, *PEXT_BLABLA;` , instead of having these typedefs kept separate down there.